packages/deno/src/executors/esbuild/esbuild.impl.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  ExecutorContext,
+  joinPathFragments,
+  logger,
+  stripIndents,
+} from '@nx/devkit';
+import * as chalk from 'chalk';
+import { dirname, join, resolve } from 'path';
+import { BuildExecutorSchema } from './schema';
+
+import { ensureDirSync, unlinkSync, writeFileSync } from 'fs-extra';
+import { CopyAssetsHandler } from '@nx/js/src/utils/assets/copy-assets-handler'
+import { processCommonArgs } from '../../utils/arg-utils';
+import { assertDenoInstalled, runDeno } from '../../utils/run-deno';
+
+export async function denoEsbuildExecutor(
+  options: BuildExecutorSchema,
+  context: ExecutorContext
+) {
+  assertDenoInstalled();
+  const opts = normalizeOptions(options, context);
+  const args = createArgs(opts, context);
+
+  const projectRoot = context.projectGraph.nodes[context.projectName].data.root;
+  const outputPath = dirname(opts.outputFile);
+
+  const assetHandler = new CopyAssetsHandler({
+    projectDir: projectRoot,
+    rootDir: context.root,
+    outputDir: outputPath,
+    assets: opts.assets,
+  });
+
+  const esbuildResult = await new Promise<boolean>((resolve) => {
+    const runningDenoProcess = runDeno(args);
+
+    runningDenoProcess.on('exit', (code) => {
+      resolve(code === 0);
+    });
+  });
+
+  if (esbuildResult !== true) {
+    return { success: false };
+  }
+
+  let copyAssetsResult = true;
+  try {
+    await assetHandler.processAllAssetsOnce();
+  } catch {
```

Comment:
Just thinking here, wonder if it would make sense to log an error saying the assets weren't able to be copied. 
idk what kind of errors this could throw but could be useful to at least log the error if verbose is being used.
```suggestion
  } catch (e) {
    if(process.env.NX_VERBOSE_LOGGING === 'true') {
      console.error(e)
    }
```